static/src/javascripts/projects/common/modules/commercial/badges.js

```diff
@@ -69,8 +69,13 @@ define([
 
             return new Promise(function (resolve) {
                 idleFastdom.write(function () {
-                    $('.js-container__header', container)
-                        .after($adSlot);
+                    var placeholder = $('.js-badge-placeholder', container);
+
+                    if (placeholder[0]) {
+                        placeholder.replaceWith($adSlot[0]);
```

Comment:
yeah definitely. I guess I was bugged by the test, it's the first time I see it this way rather than testing the value of `placeholder.length`. On a sidenote: I've never actually tested if bonzo would detect that you pass a `Node` to `replaceWith` and thus needs to clone it, otherwise the same node will just move around and end up in the last selected node.